xarray/core/interpolate.py

```diff
@@ -0,0 +1,95 @@
+"""Define core operations for xarray objects.
+
+TODO(shoyer): rewrite this module, making use of xarray.core.computation,
+NumPy's __array_ufunc__ and mixin classes instead of the unintuitive "inject"
+functions.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import pandas as pd
+
+
+def interp_along_dim(da, dim=None, method='linear', inplace=False, **kwargs):
+    '''Interpolate values according to different methods.'''
+    from .dataarray import DataArray
+
+    data = da.data
+    arr = data if inplace else data.copy()
+
+    if dim is None:
+        raise NotImplementedError('dim is required')
+    axis = da.get_axis_num(dim)
+
+    # get index
+    if method == 'linear':
+        index = np.arange(arr.shape[axis])
+    if method in ['values', 'index']:
+        index = da.get_index(dim)
+        method = 'linear'
+    elif method == 'time':
+        index = da.get_index(dim)
+        method = 'linear'
+        assert np.issubdtype(index.dtype, np.datetime64)
+    else:
+        index = da.get_index(dim)
+
+    # check index sorting now so we can skip it later
+    if not (np.diff(index) > 0).all():
+        raise ValueError("Index must be monotonicly increasing")
+
+    # TODO: consider using the np_interp_wrapper for linear method
+    out = np.apply_along_axis(scipy_interp_wrapper, axis, arr, index,
```

Comment:
I think `scipy.interpolate.interp1d` can interpolate a multiple dimensional array along one axis, though scipy is an optional requirement.
(But I did not check it is certainly faster.)

Reply:
~I think `scipy.interpolate.interp1d` can interpolate a multiple dimensional array along one axis, though scipy is an optional requirement.
(But I did not check it is certainly faster.)~

EDIT:
I noticed that I did not understand correctly.
The multi-dimensional interpolation does not work for filling the missing values which are distributed randomly.

So ignore my comment.

Reply:
yes, exactly. For 1 and 2d arrays, this should behave identically to the pandas interpolate method.

Reply:
Maybe apply_ufunc with vectorize=True would work here? I guess it would reorder dimensions in a non-desired way.

Reply:
That is on my list of things to try today. Getting things to behave like pandas required a bit of dancing and I've saved the obvious optimizations for later. Specifically, I want to get some more tests written first.